modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/analytics-view.js

```diff
@@ -17,17 +17,26 @@
  */
 
 function onRequest(context) {
+    var devices = context.unit.params.devices;
     var deviceType = context.uriParams.deviceType;
     var deviceId = request.getParameter("deviceId");
 
-    if (deviceType != null && deviceType != undefined && deviceId != null && deviceId != undefined) {
+    if (devices) {
+        return {
+            "devices": stringify(devices),
+            "backendApiUri": devicemgtProps["httpsURL"] + "/android_sense/stats/"
+        };
+    } else if (deviceType != null && deviceType != undefined && deviceId != null && deviceId != undefined) {
         var deviceModule = require("/app/modules/device.js").deviceModule;
         var device = deviceModule.viewDevice(deviceType, deviceId);
         if (device && device.status != "error") {
```

Comment:
How bracket notation could be more reliable? It is all about the language semantics. Using dot notation is permitted in JS as long as attribute name is a valid identifier.
http://stackoverflow.com/a/4968448/1391687